### DISCUSSION TITLE
How to store content in 2 (or more) languages?

### DISCUSSION THREAD

**Discussion by welcoMattic** [Cecilapp/Cecil#1169] [Q&A] [answered]
I need to let my publishers write content in 2 (and soon more) languages, but I can't find a way to load the right content.
To be more precise, the current locale will be in the url: https://mywebsite/[LOCALE]/some/path.

Is there a way to achieve that with Cecil? 

Thanks

**Comment by welcoMattic** [comment, marked as answer]
Found : https://cecil.app/documentation/content/#localized-content

**Reply by welcoMattic** [reply]
I found how to store content by languages, but not how to build a language switcher, and how to build correctly the menu for the current languages. Can we discuss about that? Thanks

**Reply by ArnaudLigny** [reply]
Hello @welcoMattic, as mentioned in the documentation localized content is an _experimental_ feature and have some limitations.

You can link translated pages between them with the `langref` variable, with something like that:

```twig
{% for p in site.pages|filter_by('langref', page.langref) %}
- <a href="{{ url(p) }}">{{ p.title }}</a><br>
{% endfor %}
```

About menu, there is no built-in helper to easily display it for a specific language: you should create a specific menu by language.

For example, in your pages' front matter :
```yaml
---
menu: main_fr
---
```

And in your template:
```twig
{% if attribute(_context, site.menus.main~'_'~site.language) is defined %}
{% for menu in attribute(_context, site.menus.main~'_'~site.language)|sort_by_weight %}
          <a class="sidebar-nav-item{% if url(page) == url(menu.url) %} active{% endif %} item-weight-{{ menu.weight }}" href="{{ url(menu.url) }}">{{ menu.name }}</a>
{% endfor %}
{% endif %}
```

**Comment by welcoMattic** [comment]
Thank you @ArnaudLigny I'll check that, and contribute to Cecil if I find some ways to improve localization feature :+1:

**Comment by ArnaudLigny** [comment]
A better documentation here: https://cecil.app/documentation/content/#multilingual.
